src/portal/lib/src/image-name-input/image-name-input.component.ts

```diff
@@ -0,0 +1,103 @@
+import { Component, OnDestroy, OnInit } from "@angular/core";
+import { Project } from "../project-policy-config/project";
+import { Observable, Subject } from "rxjs/index";
+import { debounceTime, distinctUntilChanged } from "rxjs/operators";
+import { ProjectService } from "../service/project.service";
+import { AbstractControl, FormBuilder, FormGroup, Validators } from "@angular/forms";
+import { ErrorHandler } from "../error-handler/error-handler";
+
+@Component({
+    selector: "hbr-image-name-input",
+    templateUrl: "./image-name-input.component.html",
+    styleUrls: ["./image-name-input.component.scss"]
+})
+export class ImageNameInputComponent implements OnInit, OnDestroy {
+    noProjectInfo = "";
+    selectedProjectList: Project[] = [];
+    proNameChecker: Subject<string> = new Subject<string>();
+    imageNameForm: FormGroup;
+
+    constructor(
+        private fb: FormBuilder,
+        private errorHandler: ErrorHandler,
+        private proService: ProjectService,
+    ) {
+        this.imageNameForm = this.fb.group({
+            projectName: ["", Validators.required],
+            repoName: ["", Validators.required],
+            tagName: ["", Validators.required],
+        });
+    }
+    ngOnInit(): void {
+        this.proNameChecker
+            .pipe(debounceTime(500))
```

Comment:
why here need to use debounce?

Reply:
https://www.learnrxjs.io/operators/filtering/debouncetime.html

> Reducing number of API Calls to your server using debounceTime